File: enable-big-blocks.js
const fetch = require('node-fetch');
const { createHash } = require('crypto');
const ethers = require('ethers');
const dotenv = require('dotenv');

// .env.nft dosyasını yükle
dotenv.config({ path: '.env.nft' });

// Hyperliquid API URL
const apiUrl = process.env.API_URL || 'https://api.hyperliquid.xyz';
const apiWalletName = process.env.API_WALLET_NAME || 'app.hyperliquid.xyz';
const apiWalletAddress = process.env.API_WALLET_ADDRESS;
const privateKey = process.env.PRIVATE_KEY;
const eip1193URL = process.env.RPC_URL || 'https://rpc.hyperliquid.xyz/evm';

// Private key ile bir wallet oluştur
const wallet = new ethers.Wallet(`0x${privateKey}`);

/**
 * Slow Blocks modunu etkinleştir
 */
async function enableBigBlocks() {
  try {
    console.log('Slow Blocks (Big Blocks) modu etkinleştiriliyor...');
    
    // Doğrudan RPC isteği deneyelim
    const response = await fetch(eip1193URL, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        jsonrpc: '2.0',
        id: 1,
        method: 'eth_sendTransaction',
        params: [{
          from: wallet.address,
          to: "0x0000000000000000000000000000000000000001", // Precompile address for L1Actions
          data: ethers.AbiCoder.defaultAbiCoder().encode(
            ["tuple(string, bool)"],
            [["evmUserModify", true]]
          )
        }]
      }),
    });
    
    const result = await response.json();
    console.log('RPC yanıtı:', result);
    
    if (result.error) {
      console.error('Hata:', result.error.message);
    } else {
      console.log('Slow Blocks modu etkinleştirme işlemi gönderildi!');
      console.log('İşlem Hash:', result.result);
    }
    
  } catch (error) {
    console.error('Hata:', error.message);
    
    // Alternatif yöntem deniyoruz
    try {
      console.log('\nAlternatif yöntem deneniyor...');
      
      // Web3 kullanarak
      const Web3 = require('web3');
      const web3 = new Web3(eip1193URL);
      
      // Private key ile hesap oluştur
      const account = web3.eth.accounts.privateKeyToAccount(`0x${privateKey}`);
      web3.eth.accounts.wallet.add(account);
      
      // L1 eylemi için özel data
      const data = web3.eth.abi.encodeParameters(
        ['string', 'bool'],
        ['evmUserModify', true]
      );
      
      // İşlem gönder
      const tx = await web3.eth.sendTransaction({
        from: account.address,
        to: "0x0000000000000000000000000000000000000001", // L1Action precompile address
        data: data,
        gas: 500000
      });
      
      console.log('İşlem başarılı!', tx.transactionHash);
      
    } catch (altError) {
      console.error('Alternatif yöntem hatası:', altError.message);
      console.log('\nLütfen Hyperliquid UI\'dan Big Blocks modunu manuel olarak etkinleştirin:');
      console.log('1. https://app.hyperliquid.xyz/ adresine gidin');
      console.log('2. Cüzdanınızı bağlayın');
      console.log('3. Ayarlar bölümünden "Enable Big Blocks" seçeneğini bulun');
    }
  }
}

// Slow Blocks modunu etkinleştir
enableBigBlocks();

File: enable_big_blocks.py
import argparse
from dotenv import load_dotenv
import eth_account
import os
import requests

from hyperliquid.utils.signing import (
    get_timestamp_ms,
    sign_l1_action,
)

# .env.nft dosyasını yükle
load_dotenv(dotenv_path='.env.nft')

def main(using_big_blocks):
    # .env.nft'den özel anahtarı al (0x olmadan)
    private_key = os.getenv("PRIVATE_KEY")
    if not private_key:
        raise ValueError("PRIVATE_KEY bulunamadı, lütfen .env.nft dosyasını kontrol edin.")
    
    print(f"Özel anahtar kullanılıyor: {private_key[:4]}...{private_key[-4:]}")
    
    # Wallet oluştur
    wallet = eth_account.Account.from_key(private_key)
    print(f"Cüzdan adresi: {wallet.address}")

    # Zaman damgası oluştur
    timestamp = get_timestamp_ms()
    print(f"Zaman damgası: {timestamp}")

    # L1 eylemi hazırla
    action = {
        "type": "evmUserModify",
        "usingBigBlocks": using_big_blocks
    }
    print(f"Ayarlanacak değer: usingBigBlocks={using_big_blocks}")

    # Eylemi imzala (True=Mainnet için, False=Testnet için)
    # RPC_URL içinde 'testnet' geçiyor mu kontrol et
    rpc_url = os.getenv("RPC_URL", "").lower()
    is_mainnet = "testnet" not in rpc_url
    
    print(f"Ağ: {'MAINNET' if is_mainnet else 'TESTNET'}")
    
    signature = sign_l1_action(
        wallet,
        action,
        None,
        timestamp,
        is_mainnet,  # True=Mainnet, False=Testnet
    )

    # İstek payload'ını hazırla
    payload = {
        "action": action,
        "nonce": timestamp,
        "signature": signature
    }
    print(f"Gönderilecek veri: {payload}")

    # API URL'ini belirle
    api_url = "https://api.hyperliquid.xyz/exchange" if is_mainnet else "https://api.hyperliquid-testnet.xyz/exchange"
    print(f"API URL: {api_url}")

    # POST isteği gönder
    print(f"İstek gönderiliyor...")
    response = requests.post(
        api_url,
        headers={"Content-Type": "application/json"},
        json=payload
    )

    # Yanıtı yazdır
    print(f"Durum Kodu: {response.status_code}")
    try:
        print(f"Yanıt: {response.json()}")
        
        if response.status_code == 200 and response.json().get('status') == 'ok':
            state_text = "ETKİNLEŞTİRİLDİ" if using_big_blocks else "DEVRE DIŞI BIRAKILDI"
            print(f"\n✅ Big Blocks modu başarıyla {state_text}!")
            print(f"   Artık büyük kontratları {'deploy edebilirsiniz' if using_big_blocks else 'deploy etmek için Big Blocks modunu tekrar etkinleştirmeniz gerekecek'}.")
        else:
            print(f"\n❌ Big Blocks modunu değiştirme işlemi başarısız oldu!")
            print(f"   Hata detayları: {response.json()}")
    except ValueError:
        print(f"API yanıtı JSON formatında değil: {response.text}")

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Big Blocks modunu etkinleştirme veya devre dışı bırakma scripti.")
    parser.add_argument("usingBigBlocks", type=str, choices=["true", "false"], help="Big Blocks modu için boolean değeri (true/false)")
    args = parser.parse_args()

    # String argümanı boolean'a çevir
    using_big_blocks = args.usingBigBlocks.lower() == "true"

    main(using_big_blocks)
